src/main/java/uk/gov/pay/directdebit/webhook/gocardless/services/handlers/UnhandledGoCardlessEventsLogger.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.pay.directdebit.webhook.gocardless.services.handlers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.directdebit.events.model.GoCardlessEvent;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import static uk.gov.pay.directdebit.events.model.GoCardlessEvent.ACTION_MANDATE_REPLACED;
+import static uk.gov.pay.directdebit.events.model.GoCardlessEvent.ACTION_PAYMENT_RESUBMISSION_REQUESTED;
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.MANDATES;
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.PAYMENTS;
+
+public class UnhandledGoCardlessEventsLogger {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UnhandledGoCardlessEventsLogger.class);
+
+    private static Predicate<GoCardlessEvent> byValidEventsForMandateResourceTypes =
+            event -> MANDATES.equals(event.getResourceType());
+
+    private static Predicate<GoCardlessEvent> byValidEventsForPaymentsResourceTypes =
+            event -> PAYMENTS.equals(event.getResourceType());
+    
+    public static void logUnhandledEvents(List<GoCardlessEvent> events) {
+        events.stream().filter(byValidEventsForMandateResourceTypes).forEach(UnhandledGoCardlessEventsLogger::logErrorForUnexpectedMandateEvents);
+        events.stream().filter(byValidEventsForPaymentsResourceTypes).forEach(UnhandledGoCardlessEventsLogger::logErrorForUnexpectedPaymentEvents);
+    }
+    
+    private static void logErrorForUnexpectedMandateEvents(GoCardlessEvent event) {
+        if (event.getAction().equals(ACTION_MANDATE_REPLACED)) {
```

Comment:
We also want `"customer_approval_granted"`, `"customer_approval_skipped"` and `"resubmission_requested"`.

Reply:
ah did you just add those to the acceptance criteria?

Reply:
No, they’ve been there since the beginning! Also, we’ve always been at war with Eurasia, 2 + 2 = 5, war is peace, freedom is slavery, ignorance is strength.